exotations/task_maps/task_map/include/task_map/JointAccBackFiniteDiff.h

```diff
@@ -0,0 +1,78 @@
+/*
+ *      Author: Christopher E. Mower
+ *
+ * Copyright (c) 2018, University Of Edinburgh
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ *  * Redistributions of source code must retain the above copyright notice,
+ *    this list of conditions and the following disclaimer.
+ *  * Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *  * Neither the name of  nor the names of its contributors may be used to
+ *    endorse or promote products derived from this software without specific
+ *    prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ */
+
+#ifndef JOINTACCBACKFINITEDIFF_H_
+#define JOINTACCBACKFINITEDIFF_H_
+
+#include <exotica/TaskMap.h>
+#include <task_map/JointAccBackFiniteDiffInitializer.h>
+
+namespace exotica
+{
+/**
+   * \brief	Implementation of joint acceleration estimation using backward finite differencing task map.
+   */
+
+class JointAccBackFiniteDiff : public TaskMap, public Instantiable<JointAccBackFiniteDiffInitializer>
+{
+public:
+    JointAccBackFiniteDiff() {}
+    ~JointAccBackFiniteDiff() {}
+    virtual void Instantiate(JointAccBackFiniteDiffInitializer& init);
+
+    void Initialize();
+
+    virtual void assignScene(Scene_ptr scene);
+
+    void initPrevJointState(Eigen::VectorXdRefConst joint_state);
+    void setPrevJointState(Eigen::VectorXdRefConst joint_state);
+
+    virtual void update(Eigen::VectorXdRefConst x, Eigen::VectorXdRef phi);
+
+    virtual void update(Eigen::VectorXdRefConst x, Eigen::VectorXdRef phi, Eigen::MatrixXdRef J);
+
+    virtual int taskSpaceDim();
+
+private:
+    int N;
+    Eigen::MatrixXd Q_;
```

Comment:
Please add doxygen descriptions right behind the variables of what they do/stand for.